### PR TITLE
Recommend namespacing pragmas and annotations

### PR DESCRIPTION
**Pull request by hodgestar**
### Summary

This adds a recommendation to the directives specification suggesting that implementations namespace their pragmas and annotations to avoid collisions.

It reserves the use of the `openqasm` namespace for future use by the OpenQASM specification.

It extends the allowed annotations keywords to allow a dotted list of identifiers rather than a single identifier. At the cost of adjusting the parser to be a little more permissive, this allows pragmas and annotations to use identically namespacing that follows common conventions (e.g. Java and Python import namespacing). Using dot as the name separator allows underscore to be used within a name to separate words.

### Details and comments

Questions:

- Are we happy with extending the allowed annotation keywords as described? Does this require an OpenQASM specification version bump?
- Are we happy with the examples in the specification? I wasn't sure whether using `ibm.` was appropriate, but I also didn't want to make the examples too abstract.
- I would like to maintain a table of known namespaces, but the specification itself didn't feel like the correct place. Should I create a file somewhere else? If so, where? Or just add a small table to the specification for now?

### CODE REVIEW

File: source/language/directives.rst [hunk 1]
```diff
@@ -40,10 +70,10 @@ the entire circuit.
    OPENQASM 3.0;
 
    // Attach billing information
-   pragma user alice account 12345678
+   pragma ibmq.user alice account 12345678
```
Comment (hodgestar) [inline source/language/directives.rst:73]:
```suggestion
   pragma ibm.user alice account 12345678
```

File: source/language/directives.rst [hunk 2]
```diff
@@ -40,10 +70,10 @@ the entire circuit.
...
 
    // Assert that the QPU is healthy to run this circuit
-   pragma max_temp qpu 0.4
+   pragma ibmq.max_temp qpu 0.4
```
Comment (hodgestar) [inline source/language/directives.rst:76]:
```suggestion
   pragma ibm.max_temp qpu 0.4
```

**Comment by braised-babbage** [comment]
I'm supportive of this idea, and taken on its own it seems pretty safe. My only request is that we consider how this comports with other potential notions of namespacing (e.g. as described in https://github.com/openqasm/openqasm/issues/365). For example, are there any caveats around the use of `.` as a separator more broadly (if we introduce namespaces as a notion outside of pragmas & annotations)? Is this even something we should be considering here?

**Review by blakejohnson — APPROVED** [review]
I feel like this is sufficiently contained that we can move forward without solving for namespacing in the broader language.

**Review by levbishop — APPROVED** [review]

**Pull request merged**